### PR TITLE
test(native): Add e2e tests for FNV1 hash functions

### PR DESCRIPTION
**Pull request by anandamideShakyan**
## Description
Add e2e tests for fnv1_32, fnv1_64, fnv1a_32 and fnv1a_64 functions.
## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.
- [ ] If adding new dependencies, verified they have an [OpenSSF Scorecard](https://securityscorecards.dev/#the-checks) score of 5.0 or higher (or obtained explicit TSC approval for lower scores).

## Release Notes

```
== NO RELEASE NOTE ==
```

## Summary by Sourcery

Add shared end-to-end test coverage for FNV hash binary functions across Java and native implementations.

Tests:
- Introduce an abstract binary functions test suite validating fnv1_32, fnv1_64, fnv1a_32, and fnv1a_64 behavior, edge cases, aggregation usage, and binary data handling.
- Add Java-based and native C++-backed concrete test runners to execute the shared binary function tests against both implementations.

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import org.testng.annotations.Test;
+
+/**
+ * Abstract test class for binary functions including FNV hash functions.
+ * This class contains E2E tests that can be run against both Java and native implementations.
+ */
```
Comment (pdabre12) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java]: I think we just run these tests against the same query runner, we should add some tests comparing both Java and C++ results.

Comment (anandamideShakyan) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java]: But we do run the same set of tests(with same expected outputs) for both java and cpp. TestBinaryFunctionsNative tests for cpp and TestBinaryFunctions tests for java. Since we run same tests on both so I think this is sufficient. Wdyt ?

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+{
+    /**
+     * Test FNV-1 32-bit hash function.
+     * Ground truth results are generated via https://nqv.github.io/fnv/
+     */
+    @Test
+    public void testFnv1_32()
+    {
+        // Test empty input - should return offset basis
+        assertQueryWithSameQueryRunner("SELECT fnv1_32(from_hex(''))", "SELECT -2128831035");
```
Comment (czentgr) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java]: So the question about the query runner, usually we have the expected query runner and a query runner.
The expected query runner for native is the Java query runner as seen in the other native tests.
In this case we compare to the literal as also pointed out by the AI review. We don't compare Java and C++. We compare against expected literals.

Now even with sidecar this is currently constant folded (eventually not).
So easiest would be to have some mix of constant folded and values clauses maybe to exercise both paths.

Comment (anandamideShakyan) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestBinaryFunctions.java]: The reason for using AssertQuerywithSameQueryRunner is that h2 queryrunner doesn't contain presto functions.... so it was throwing following "Function not found error", but since I don't call the fnv functions in expected so I think I can use AssertQuery as well. I will add some tests with Values clause as well.
Screenshot 2026-05-21 at 10 57 10 AM

File: presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctionsNative.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import static com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils.nativeHiveQueryRunnerBuilder;
+import static com.facebook.presto.sidecar.NativeSidecarPluginQueryRunnerUtils.setupNativeSidecarPlugin;
+import static java.lang.Boolean.parseBoolean;
+
+/**
+ * E2E test suite for binary functions including FNV hash functions.
+ * This test runs against the native C++ implementation using PrestoNativeQueryRunner.
+ *
+ * The tests verify that the C++ implementation produces identical results to the Java
+ * implementation for all FNV hash variants (fnv1-32, fnv1-64, fnv1a-32, fnv1a-64).
```
[the remark this answers is not in the document]
Comment (czentgr) [reply presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctionsNative.java]: Maybe we can reword this better because we don't actually run the comparison to the different runners.

File: presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctionsNative.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class TestBinaryFunctionsNative
```
Comment (czentgr) [inline presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctionsNative.java]: We don;t need to say `TestBinaryFunctionsNative`. They are already in the native package. `TestBinaryFunctions` is sufficient.

Comment (anandamideShakyan) [reply presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctionsNative.java]: Done.

**Review by czentgr — APPROVED** [review]
Please rebase.
@aditi-pandit Please review.

**Review by aditi-pandit — APPROVED** [review]
Thanks @anandamideShakyan for these tests !

**Review by aditi-pandit — CHANGES_REQUESTED** [review]
Hi @anandamideShakyan : Can we avoid the test files in presto-tests and move them to presto-native-tests itself ?

**Comment by anandamideShakyan** [comment]
> Hi @anandamideShakyan : Can we avoid the test files in presto-tests and move them to presto-native-tests itself ?

Hi @aditi-pandit. That's not possible since we are using the same AbstractTest file to test out java(presto-tests/src/test/java/com/facebook/presto/tests/TestBinaryFunctions.java) as well as native implementation (presto-native-tests/src/test/java/com/facebook/presto/nativetests/TestBinaryFunctions.java), also other e2e tests follow the same pattern.

**Comment by czentgr** [comment]
@aditi-pandit The E2E tests are completely missing for these functions. So this is adding them for Java and native at the same time. I think this is fine.

**Review by aditi-pandit — APPROVED** [review]
So presto-tests are tested with Java vs H2 query runners in addition to the native setup. I was trying to minimize the java testing with the focus on C++.

I am fine with having these new tests as first class in presto-native-tests only without presto-tests addition. But I don't want to block this PR on that either.

@czentgr

**Review by tdcmeehan — APPROVED** [review]
